contracts/DCAHubCompanion/DCAHubCompanionTakeWithdrawAndSwapHandler.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.8.7 <0.9.0;
+
+import '@mean-finance/swappers/solidity/contracts/SwapAdapter.sol';
+import '../interfaces/IDCAHubCompanion.sol';
+
+/// @dev All public functions are payable, so that they can be multicalled together with other payable functions when msg.value > 0
+abstract contract DCAHubCompanionTakeWithdrawAndSwapHandler is SwapAdapter, IDCAHubCompanionTakeWithdrawAndSwapHandler {
+  using SafeERC20 for IERC20;
+  using Address for address payable;
+
+  /// @inheritdoc IDCAHubCompanionTakeWithdrawAndSwapHandler
+  function sendToRecipient(
+    address _token,
+    uint256 _amount,
+    address _recipient
+  ) external payable {
+    if (_token == PROTOCOL_TOKEN) {
+      payable(_recipient).sendValue(_amount);
+    } else {
+      IERC20(_token).safeTransfer(_recipient, _amount);
+    }
+  }
+
+  /// @inheritdoc IDCAHubCompanionTakeWithdrawAndSwapHandler
+  function takeFromCaller(IERC20 _token, uint256 _amount) external payable {
```

Comment:
I didn't add tests for `takeFromCaller` and `sendAllBalanceToRecipient`, since they are only exposing already tested functions. Is that ok for you?

Reply:
I mean, you could test the "internal call". It seems kind of "too much" but you know how much I like reducing errors scopes and increased testing lol. Up to you tb completely honest.

Reply:
Ok, will add the tests